### ISSUE TITLE
Exception 613: Calls to this api have exceeded the rate limit during custom audience upload

### ISSUE THREAD

**Issue by steveHimself1397** [facebook/facebook-php-ads-sdk#344]
Hi,

I am trying to upload a lot of smaller files with 100 to a few 1000 email adresses in them.
I wrote a script for it and it works fine on a small amount of files with lots of email adresses in them (i split them up in 10.000 adresses packages and upload them)

But if I try to upload a large amount of files (roughly 3.5k) with small amounts of email addresses in them (say roughly 1 - 10k) it throws an Exception after around 5 - 8 files.

The exception thrown has the errorCode 613 and the message: "Calls to this api have exceeded the rate limit"

Now. The only place I could find this error code in the facebook documentation was in the messenger API. Since I am not using this I don't understand how this exception gets thrown.

I've also implemented a "wait 5 minutes in case of this exception" but after the 5 minutes wait I still can't make another request and I again land in the same exception which triggers another wait and so on.

Basically - I can't upload large amount of custom audience files due to rate limits.
Any tips how to tackle this?

Thanks,
Steve

**Comment by guxufb** [comment]
Please go to https://www.facebook.com/groups/pmdcommunity for general questions regarding to API usage.
